crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -551,14 +589,15 @@ impl<Wg: WireGuard, E: Backoff> Runtime for State<Wg, E> {
                             // This is a stun timeout. We should back off and move to the next server.
                             self.transition_to_backing_off_state().await;
                         } else {
-                            // This is a poll interval. We're still in HasEndpoints state.
+                            // This is a poll interval. We're still in HasEndpoints state
                             let res = self.start_stun_session().await;
                             if let Err(err) = res {
                                 telio_log_error!("Starting STUN session failed with error: {:?}", err);
                                 self.transition_to_backing_off_state().await;
                             }
                         }
                     },
+                    StunState::Paused => (),
```

Comment:
In paused state this will form infinite loop

Reply:
Much better place to handle "pause" state would be: https://github.com/NordSecurity/libtelio/pull/168/files#diff-f52b9bdf9a9389d5ebb2456346414b982fc5fd532c9cf08e23284a430ff06003R555